worker/modelcache/worker.go

```diff
@@ -29,8 +29,14 @@ type BackingWatcher interface {
 	Stop() error
 }
 
+// Unlocker is used to indicate that the model cache is ready to be used.
+type Unlocker interface {
+	Unlock()
+}
+
 // Config describes the necessary fields for NewWorker.
 type Config struct {
+	InitializedGate      Unlocker
```

Comment:
Will tend to stutter as variables are called `gate`.

Reply:
Actually none of the variables are called `gate`. But `initialized` or `unlocker`. The later of which is a bit odd IMO.